docs/source/autograd.rst

```diff
@@ -11,6 +11,32 @@ Automatic differentiation package - torch.autograd
 
 .. autofunction:: grad
 
+.. _functional-api:
+
+Functional higher level API
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section contains the higher level API for the autograd that build on the basic API above.
+
+This API always work with user-provided functions that take only Tensors as input and return
+only Tensors.
+If your function takes other arguments that are not Tensors and that should be ignored for the
```

Comment:
This doesn't tell us what happens if we want other arguments to be Tensors but not compute their derivatives so it's a bit ambiguous. Maybe an example that captures another tensor would help here (or a rephrasing as "other arguments that are not Tensors and should be ignored, or arguments that are Tensors that should not be differentiated")